kansa/lib/siteselect.js

```diff
@@ -92,9 +92,10 @@ class Siteselect {
         if (data.used) throw new InputError(`Token already used at ${data.used}`)
         return this.db.oneOrNone(`
           SELECT p.legal_name, p.email, s.time AS vote_time
-            FROM people p LEFT JOIN siteselection_votes s ON (p.id = s.person_id)
-           WHERE p.id = $1 AND p.membership IN
-                 ('Supporter','Youth','FirstWorldcon','Adult')`, id)
+          FROM people p
+            LEFT JOIN siteselection_votes s ON (p.id = s.person_id)
+            LEFT JOIN membership_types m USING (membership)
+          WHERE p.id = $1 AND m.wsfs_member = true`, id)
```

Comment:
Magic numbers in switch case. Is there any way we could name these numbers?
Either that or a nice little comment about what they represent?

Vote seems like an important function. Could we get a little test going for it?

Reply:
Tbh I can't remember why I ended up doing transactions this way; probably misunderstood that a construction like `db.tx(t => t.one(...).then(res => t.one(...)))` wouldn't work for some reason. Should refactor all of these.